src/test/java/eu/deltasource/internship/hotel/service/RoomServiceTest.java

```diff
@@ -112,29 +123,86 @@ void saveRooms() {
 	}
 
 	@Test
-	void deleteRoom() {
+	void saveRoomsShouldThrowExcWhenTheRoomToBeSavedHasInvalidValues() {
+		//given
+		Set<AbstractCommodity> commoditiesSetWithNullMembers = new HashSet<>(Arrays.asList(new Bed(SINGLE), null, new Toilet()));
+		Room roomWithNullCommodity = new Room(1, commoditiesSetWithNullMembers);
+		//then
+		assertThrows(ItemNotFoundException.class, () -> {
+			roomService.saveRooms(roomWithNullCommodity);
+		});
+	}
+
+	@Test
+	void deleteRoomShouldReturnTrueWhenDeletionIsSuccessful() {
 		//then
 		assertTrue(roomService.deleteRoom(doubleRoom));
 		assertFalse(roomService.findRooms().contains(doubleRoom));
 	}
 
 	@Test
-	void deleteRoomById() {
-		//when
-		roomService.deleteRoomById(threePeopleKingSizeRoom.getRoomId());
+	void deleteRoomShouldThrowExcWhenRoomIsNullOrDoesNotExist(){
+		//given
+		Room nullRoom = null ;
+		Room nonExistentRoom = new Room(236,singleSet);
 		//then
+		assertThrows(ItemNotFoundException.class,()->{
+			roomService.deleteRoom(nullRoom);
+		});
+		assertThrows(ItemNotFoundException.class,()->{
+			roomService.deleteRoom(nonExistentRoom);
+		});
+	}
+
+	@Test
+	void deleteRoomByIdShouldReturnTrueWhenDeletionIsSuccessful() {
+		//then
+		assertTrue(roomService.deleteRoomById(threePeopleKingSizeRoom.getRoomId()));
 		assertFalse(roomService.findRooms().contains(threePeopleKingSizeRoom));
 	}
 
 	@Test
-	void updateRoom() {
+	void deleteRoomByIdShouldThrowExcWhenIdIsOutOfBoundsOrDoesntExist(){
+		//then
+		assertThrows(ItemNotFoundException.class,()->{
+			roomService.deleteRoomById(-1);
+		});
+		assertThrows(ItemNotFoundException.class,()->{
+			roomService.deleteRoomById(250);
+		});
+	}
+
+	@Test
+	void updateRoomShouldMakeChangesToTheRoomWithTheSpecifiedRoomId() {
 		//given
 		Room testRoom = new Room(roomService.getRoomById(1).getRoomId(), singleSet);
 		//when
 		roomService.updateRoom(testRoom);
 		//then
 		assertEquals(testRoom.getRoomCapacity(), roomService.getRoomById(1).getRoomCapacity());
+		assertEquals(testRoom.getCommodities(), roomService.getRoomById(1).getCommodities());
 
 	}
+
+	@Test
+	void updateRoomShouldThrowExceptionWhenThePassedRoomObjectHasInvalidValues() {
+		//given
+		Set<AbstractCommodity> commodities = new HashSet<>(Arrays.asList(new Bed(SINGLE), null, new Toilet()));
+		Room nullRoom = null;
+		Room notPresentRoom = new Room(234, singleSet);
+		Room roomWithNullCommodity = new Room(doubleRoom.getRoomId(), commodities); // doubleRoom already exists in the room service's repository.
+		roomRepository.save(roomWithNullCommodity);
+		//then
+		assertThrows(ItemNotFoundException.class, () -> {
+			roomService.updateRoom(nullRoom);
+		});
+
+		assertThrows(ItemNotFoundException.class, () -> {
+			roomService.updateRoom(notPresentRoom);
+		});
+		assertThrows(ItemNotFoundException.class, () -> {
+			roomService.updateRoom(roomWithNullCommodity);
```

Comment:
Doesn't make much sense to throw an ItemNotFound when an existing room is invalid.